internal/executor/clone.go

```diff
@@ -169,7 +184,7 @@ func CloneRepository(
 		return false
 	}
 
-	if ref.Hash() != plumbing.NewHash(change) {
+	if !useMergeRef && ref.Hash() != plumbing.NewHash(change) {
```

Comment:
Maybe add a log line that explains that hard reset is skipped for the PR due to resolution strategy?

Reply:
Added in https://github.com/cirruslabs/cirrus-ci-agent/pull/295/commits/bf5e7d88b559f534dbbfa3b840a71ce947f7a03a.